feat: LightClapper node

marianarafaelwhite opened the pull request:
This change introduces the LightClapper node.

At this point, the LightClapper SW can correctly
interface with HW, send data to a ThingSpeak
channel, and can be read back into a DB.

marianarafaelwhite committed:
feat: LightClapper node

This change introduces the LightClapper node.

At this point, the LightClapper SW can correctly
interface with HW, send data to a ThingSpeak
channel, and can be read back into a DB.

marianarafaelwhite commented on lightclapper/lightclapperclient.py:
```diff
@@ -0,0 +1,134 @@
+
+            data['date'] = date_list[0]
+            data['time'] = date_list[1]
+            data['lightStatus'] = f['field1']
```
Should update this later to be f.get('field1', None). Then do a quick check that the value is valid or exists (i.e. not None)

marianarafaelwhite commented on lightclapper/lightclapper.py:
```diff
@@ -0,0 +1,154 @@
+        """
+        Write status of light clapper to channel
+        """
+        status = 1 if self.__led.get_status() else 0
```
Magic numbers; create constants to represent 1 and 0

marianarafaelwhite commented on lightclapper/lightclapperclient.py:
```diff
@@ -0,0 +1,134 @@
+            if len(date_list) < 2:
+                return None
+
+            data['date'] = date_list[0]
```
Magic numbers here. Either add a comment to explain this better, or constants to show where 0 and 1 come from

marianarafaelwhite commented on lightclapper/sqliteDB.py:
```diff
@@ -0,0 +1,248 @@
+"""
+db.py
```
Update file name in comments.

marianarafaelwhite commented on lightclapper/constants.py:
```diff
@@ -0,0 +1,21 @@
+L2_M_5C2_FEED = '1208490'
+
+# URL Syntax for LightClapper
+READ_URL = 'https://api.thingspeak.com/channels/' + \
```
You can maybe experiment with updating this URL to be in the form of api.thingspeak.com/channels/CHANNEL+FEED/feeds.json?api_key=READ_KEY

marianarafaelwhite reviewed (commented):
Minor comments; could be changed later

Hasan-Baig reviewed (approved):
Looks good to commit: 
- constants.py might move to a database or keep as is

marianarafaelwhite committed:
Change ThingSpeak Client key

Corrected the channel information for the
light clapper clients's thingspeak read operation

marianarafaelwhite committed:
fix: update magic numbers, Thingspeak

This commit replaces any magic numbers,
updated ThingSpeak to read all data,
added notice that docstrings use numpydoc.

To do:
 - udpate test cases
 - add more error handling
 - move common files to common folder

marianarafaelwhite committed:
fix: update LightClapperClient

Fix LightClapperClient to read back only
new data, except for on first pass where
there is more overhead.

Minor fixes and improvements throughout.
Documentation updated in unittests.

marianarafaelwhite committed:
feat: LightClapperDB get_records()

Introduced get_records() method for LightClapperDB
Introduced manual test for LightClapperDB
Minor updates throughout files.

Next: add unit tests for LightClapperDB

marianarafaelwhite committed:
feat: LightClapperDB tests

Introduce LightClapperDB unittests
Minor updates throughout
New bash script to run unittests for LightClapper

marianarafaelwhite committed:
fix: Improve LightClapper for Demo2

Minor code improvements throughout.
- use of enums for status of toggle, mic, etc.
- README for unit tests DB
- comments and docstring updates
- check for lightStatus as 0 or 1
- re-arranging of code in if/else statements

marianarafaelwhite commented:
Updated with more commits for review before merging with main branch.

marianarafaelwhite committed:
feat: new LightClapper info log for demo2

marianarafaelwhite committed:
fix: updated TestLightClapper method names

change on/off to toggled/not_toggled
